Incremental PDF loading code should build on iOS family

aprotyas opened the pull request:
#### aab2b2258525f0d9ab9a188fbf45ad9c3f7e600b
<pre>
Incremental PDF loading code should build on iOS family
<a href="https://bugs.webkit.org/show_bug.cgi?id=286569">https://bugs.webkit.org/show_bug.cgi?id=286569</a>
<a href="https://rdar.apple.com/143686579">rdar://143686579</a>

Reviewed by Megan Gardner and Wenson Hsieh.

This patch sets HAVE_INCREMENTAL_PDF_APIS for PLATFORM(IOS_FAMILY),
which allows incremental PDF loading code to build on the iOS family.
Note that all requisite CoreGraphics and PDFKit interfaces are available
on iOS already.

We write this patch in anticipation of webkit.org/b/286549, which tracks
_enablement_ of the incremental PDF loading behavior on iOS.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:

Move `-[PDFDocument resetFormFields:]` out into an existing category for
annotations. Also, introduce an IncrementalLoading category for relevant
PDFDocument methods.

Furthermore, given these methods have long existed in internal SDKs, we
move the forward declarations inside `!USE(APPLE_INTERNAL_SDK)`,
allowing internal builds to pick up the signatures from PDFDocumentPriv.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:

Canonical link: <a href="https://commits.webkit.org/289535@main">https://commits.webkit.org/289535@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aad410463e44c419462eab22ccdb3cf578941468

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87262 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6772 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41622 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92123 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/38002 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7048 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14844 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/67439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/38002 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90264 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5401 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78977 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47760 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5174 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33358 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37121 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/80051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/75657 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34232 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94009 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/86036 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14424 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/10501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/76244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14629 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74833 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75449 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18558 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19787 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18234 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/7356 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14443 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/19736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/108530 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14188 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26118 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17631 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15969 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -609,7 +609,7 @@
 #endif
 
-#if PLATFORM(MAC)
+#if PLATFORM(MAC) && PLATFORM(IOS_FAMILY)
```
I think this can just be `PLATFORM(COCOA)` now.

aprotyas replied:
True, I'll fix it.

whsieh commented on Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
```diff
@@ -98,29 +98,29 @@
 @end
 
+@interface PDFDocument (Annotations)
+-(void)resetFormFields:(PDFActionResetForm *) action;
```
(While we're moving this anyways)

```suggestion
- (void)resetFormFields:(PDFActionResetForm *)action;
```

aprotyas replied:
Will do

pxlcoder commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -609,7 +609,7 @@
 #endif
 
-#if PLATFORM(MAC)
+#if PLATFORM(MAC) && PLATFORM(IOS_FAMILY)
```
`PLATFORM(COCOA)`?

aprotyas replied:
Oh, of course!

aprotyas committed:
Incremental PDF loading code should build on iOS family
https://bugs.webkit.org/show_bug.cgi?id=286569
rdar://143686579

Reviewed by Megan Gardner and Wenson Hsieh.

This patch sets HAVE_INCREMENTAL_PDF_APIS for PLATFORM(IOS_FAMILY),
which allows incremental PDF loading code to build on the iOS family.
Note that all requisite CoreGraphics and PDFKit interfaces are available
on iOS already.

We write this patch in anticipation of webkit.org/b/286549, which tracks
_enablement_ of the incremental PDF loading behavior on iOS.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:

Move `-[PDFDocument resetFormFields:]` out into an existing category for
annotations. Also, introduce an IncrementalLoading category for relevant
PDFDocument methods.

Furthermore, given these methods have long existed in internal SDKs, we
move the forward declarations inside `!USE(APPLE_INTERNAL_SDK)`,
allowing internal builds to pick up the signatures from PDFDocumentPriv.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:

Canonical link: https://commits.webkit.org/289535@main
